src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -29,31 +29,57 @@ Copyright (c) .NET Foundation. All rights reserved.
     included in the generated deps.json.
     ============================================================
     -->
+  <UsingTask TaskName="FindNativeDeps" AssemblyFile="$(ILLinkTasksAssembly)" />
   <Target Name="_ILLink"
           Condition=" '$(PublishTrimmed)' == 'true' And
                       '$(_TargetFrameworkVersionWithoutV)' >= '3.0' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' "
           DependsOnTargets="_RunILLink">
 
+    <ItemGroup>
+      <_LinkedManagedAssemblies Include="@(_LinkedManagedAssembliesCandidates)" Condition="Exists('%(Identity)')" />
+    </ItemGroup>
+
+    <!-- Filter out native assets no longer needed after linking. -->
+    <ItemGroup Condition=" '$(TrimmerKeepNativeAssets)' != 'true' ">
+      <_NativeAssetsToAlwaysKeep Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <_NativeAssetsToAlwaysKeep Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
+      <!-- The full publish deps file path is generated after this target, so just depend on the filename. -->
```

Comment:
What about clrcompression.dll? What scenarios is it used?

Reply:
Should be keep debugshim.dll and sos.dll in debug builds?

Reply:
I believe we should keep all native binaries that you typically find in your bin folder when you build dotnet/coreclr

Reply:
Here's a list of files that i think should be copied from the runtime pack (not sure about all of them though... @swaroop-sridhar can you comment if there are files we should remove from this list?) :
clrcompression.dll
clretwrc.dll
clrjit.dll
coreclr.dll
dbgshim.dll
mscordaccore*.dll
mscordbi.dll
mscorrc.debug.dll
mscorrc.dll
sos*.dll

Reply:
I don't think we should preserve the debug DLLs by default -- at least not in release builds.
There can be a switch to control this.
I'm not sure where `clrcompression.dll` is used, so this needs better understanding.

Reply:
And ucrtbase.dll (coreclr.dll depends on this I believe)

Reply:
> I'm not sure where clrcompression.dll is used, so this needs better understanding.

If you use the Compression APIs i believe